examples/aws-nodejs/public/index.html

```diff
@@ -2,57 +2,81 @@
-            return fetch('/sign-s3', {
-              method: 'post',
-              // Send and receive JSON.
-              headers: {
-                accept: 'application/json',
-                'content-type': 'application/json',
-              },
-              body: JSON.stringify({
-                filename: file.name,
-                contentType: file.type,
-              }),
-            }).then((response) => {
-              // Parse the JSON response.
-              return response.json()
-            }).then((data) => {
-              // Return an object in the correct shape.
-              return {
-                method: data.method,
-                url: data.url,
-                fields: data.fields, // For presigned PUT uploads, this should be left empty.
-                // Provide content type header required by S3
+          .use(AwsS3, {
+            getUploadParameters (file) {
+              // Send a request to our Express.js signing endpoint.
+              return fetch('/sign-s3', {
+                method: 'post',
+                // Send and receive JSON.
                 headers: {
-                  'Content-Type': file.type,
+                  accept: 'application/json',
+                  'content-type': 'application/json',
                 },
-              }
-            })
-          },
-      });
-
-      uppy.on('complete', (result) => {
-        console.log('Upload complete! We’ve uploaded these files:', result.successful)
-      })
-
-      uppy.on('upload-success', (file, data) => {
-        console.log('Upload success! We’ve uploaded this file:', file.meta['name'])
-      })
+                body: JSON.stringify({
+                  filename: file.name,
+                  contentType: file.type,
+                }),
+              }).then((response) => {
+                // Parse the JSON response.
+                return response.json()
+              }).then((data) => {
+                // Return an object in the correct shape.
+                return {
+                  method: data.method,
+                  url: data.url,
+                  fields: data.fields, // For presigned PUT uploads, this should be left empty.
+                  // Provide content type header required by S3
+                  headers: {
+                    'Content-Type': file.type,
+                  },
+                }
+              })
+            },
+          });
+        
+        uppy.on('complete', (result) => {
+          console.log('Upload complete! We’ve uploaded these files:', result.successful)
+        })
+        
+        uppy.on('upload-success', (file, data) => {
+          console.log('Upload success! We’ve uploaded this file:', file.meta['name'])
+        })
+      }
+      {
+        const uppy = new Uppy()
+          .use(Dashboard, {
+            inline: true,
+            target: '#aws-multipart',
+          })
+          .use(AwsS3Multipart, {
+            companionUrl: window.origin,
```

Comment:
We should implement this not with `companionUrl` but with `createMultipartUpload`, `listParts`, `signPart`, `abortMultipartUpload`, and `completeMultipartUpload`. First of all because it's confusing to still use `companionUrl`, but more importantly this requires the routes to exactly match those of Companion, which could also break at some point. People most likely define their own endpoints.

Reply:
We should probably consider leave an option to avoid to have to reimplement the whole logic just to change the URL (but +1 for not keeping "companion" is the option name if we're doing this). Something to have in mind when we refactor the plugins next month.